tern/analyze/common.py

```diff
@@ -452,3 +454,100 @@ def get_os_style(image_layer, binary):
                     package_manager=binary,
                     package_format=image_layer.pkg_format,
                     os_list=image_layer.os_guess), 'info'))
+
+
+def check_git_folder(path_to_folder):
+    '''Given a path to git folder, return the name and sha of repository.
+    This function looks into .git folder.'''
+    comment_line = ''
+    if os.path.isdir(path_to_folder):
+        # find .git folder
+        dot_git_folder_path = os.path.join(path_to_folder, '.git/')
```

Comment:
You can find the top level directory (that will container .git inside it) with `git rev-parse --show-toplevel` command.

Reply:
Thanks for your suggestion!